src/main.ts

```diff
@@ -244,6 +274,21 @@ class ModuleInstance extends InstanceBase<Config> {
 			this.debouncedCheckFeedbacks(Feedback.SectionColor)
 		})
 		server.on('/setlist/activeSongName', ([, activeSongName]) => {
+			if (activeSongName) {
+				this.sectionWaveforms = []
+				let sectionsWaveData = [] as number[][]
+
+				try {
+					const data = fs.readFileSync(`waveforms/${activeSongName}.json`, 'utf8')
+					sectionsWaveData = JSON.parse(data)
+				} catch (error) {
+					this.log('error', `Failed to read or parse sections wave data: ${error}`)
+				}
+				sectionsWaveData.forEach(([id, ...values]) => {
+					this.renderWaveform(id, values.slice(0, 72), values.slice(72, 144))
+				})
```

Comment:
At this point I just read .json file, but preferabbly it to be automatically provided by osc server.